azure-storage-blob/azure/storage/blob/blob_service_client.py

```diff
@@ -88,13 +121,63 @@ def generate_shared_access_signature(
+        Use the returned signature with the sas_token parameter of any BlobService.
+        :param ResourceTypes resource_types:
+            Specifies the resource types that are accessible with the account SAS.
+        :param AccountPermissions permission:
+            The permissions associated with the shared access signature. The
+            user is restricted to operations allowed by the permissions.
+            Required unless an id is given referencing a stored access policy
+            which contains this field. This field must be omitted if it has been
+            specified in an associated stored access policy.
+        :param expiry:
+            The time at which the shared access signature becomes invalid.
+            Required unless an id is given referencing a stored access policy
+            which contains this field. This field must be omitted if it has
+            been specified in an associated stored access policy. Azure will always
+            convert values to UTC. If a date is passed in without timezone info, it
+            is assumed to be UTC.
+        :type expiry: datetime or str
+        :param start:
+            The time at which the shared access signature becomes valid. If
+            omitted, start time for this call is assumed to be the time when the
+            storage service receives the request. Azure will always convert values
+            to UTC. If a date is passed in without timezone info, it is assumed to
+            be UTC.
+        :type start: datetime or str
+        :param str ip:
+            Specifies an IP address or a range of IP addresses from which to accept requests.
+            If the IP address from which the request originates does not match the IP address
+            or address range specified on the SAS token, the request is not authenticated.
+            For example, specifying sip=168.1.5.65 or sip=168.1.5.60-168.1.5.70 on the SAS
+            restricts the request to those IP addresses.
+        :param str protocol:
+            Specifies the protocol permitted for a request made. The default value
+            is https,http. See :class:`~azure.storage.common.models.Protocol` for possible values.
+        :return: A Shared Access Signature (sas) token.
+        :rtype: str
+        '''
+        if self.account_name is None:
+            raise ValueError('Account name should not be None')
+        if self.account_key is None:
+            raise ValueError('Account key should not be None')
+
+        sas = SharedAccessSignature(self.account_name, self.account_key)
+        return sas.generate_account(Services.BLOB, resource_types, permission,
+                                    expiry, start=start, ip=ip, protocol=protocol)
 
     def get_account_information(self, timeout=None):
         # type: (Optional[int]) -> Dict[str, str]
         """
         :returns: A dict of account information (SKU and account type).
```

Comment:
As a user I would be happy if this docstring listed the explicit key values that can be expected, or gave an example output